interp/dumpglob.mli

```diff
@@ -8,6 +8,95 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
+(** This file implements the Coq's [.glob] file format, which
+   provides information about the objects that are defined and referenced
+   from a Coq file.
+
+   The [.glob] file format is notably used by [coqdoc] and [coq2html]
+   to generate links and other documentation meta-data.
+
+   Note that we consider this format a legacy one, and no stability
+   guarantees are provided as of today, as we search to replace this
+   format with a more structured and strongly-typed API.
```

Comment:
FWIW, I've been thinking a bit about adding a "jump to definition" feature in CoqIDE (and available to other IDEs), perhaps something I'd look at later this year after further debugger work.  I was thinking in terms of a new call in the IDE protocol (passing filename and offset where the user clicked, returning the location to jump to), which could distinguish references to symbols with the same name in parsed files.

Also maybe a "search by name" feature that takes a substring or regex with dynamic updates of the result for each character the user types.

While I don't quite understand exactly how coqdoc uses the .glob info, these seem closely related.  Are you planning to create the "more structured and strongly-typed API" anytime soon?

Reply:
The relevant bit of glob files for your feature is that coqdoc will look for `R`*start*`:`*end* lines that overlap with the byte range of the identifier coqdoc is considering linking, use the kind part to know how to highlight the identifier, and then use the modpath to determine the file the reference lives in end the name and secpath to determine which identifier in that file to link to.  The anchors are created from the definition sort of lines, which say that a given byte range defines a given kind of thing with a given name and secpath.

If you want "search by name", I think `Search` with a string already does that; `Locate` will then give you the fully qualified name of whatever you're looking for, which can then be used to determine what file and what identifier in that file to jump to.

Reply:
IIUC, coqdoc generates html files (such as for the standard library) and the .glob files, but I don't see how the HTML uses the .glob files at all--it just creates HTML tags like the following (in Coq.Init.Datatypes.html).  What am I missing here?

![image](https://user-images.githubusercontent.com/1253341/150179316-a4b8f7fa-be1e-4978-b18c-cf228b1d60c7.png)

> If you want "search by name", I think `Search` with a string already does that; `Locate` will then give you the fully qualified name of whatever you're looking for, which can then be used to determine what file and what identifier in that file to jump to.

Perhaps.  But it's not very convenient for users to have to add two commands to their script and then manually navigate to the appropriate file and line number, then remove the two commands.  Much easier to use as a GUI feature.

Reply:
> IIUC, coqdoc generates html files (such as for the standard library) and the .glob files

coqc generates the .glob files (and also .vo files, by consuming .v files).  coqdoc consumes .glob files (combined with the corresponding .v file) to generate .html files.

Reply:
> But it's not very convenient for users to have to add two commands to their script and then manually navigate to the appropriate file and line number, then remove the two commands. Much easier to use as a GUI feature.

Right, I'm suggesting that the GUI could use the API that already exists for `Search` and `Locate` to perform this operation for the user.

Reply:
The XML protocol actually did have a way to forward the Glob info, it survived by chance tho :)